components/log_wrappers/src/lib.rs

```diff
@@ -86,7 +101,67 @@ impl<'a> slog::Value for Key<'a> {
         key: slog::Key,
         serializer: &mut dyn slog::Serializer,
     ) -> slog::Result {
-        serializer.emit_arguments(key, &format_args!("{}", hex::encode_upper(self.0)))
+        if REDACT_INFO_LOG.load(Ordering::Relaxed) {
+            serializer.emit_arguments(key, &format_args!("<key>"))
+        } else {
+            serializer.emit_arguments(key, &format_args!("{}", hex::encode_upper(self.0)))
+        }
+    }
+}
+
+impl<'a> fmt::Display for Key<'a> {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if REDACT_INFO_LOG.load(Ordering::Relaxed) {
+            // Print placeholder instead of the key itself.
+            write!(f, "<key>")
+        } else {
+            write!(f, "{}", hex::encode_upper(self.0))
+        }
+    }
+}
+
+impl<'a> fmt::Debug for Key<'a> {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(self, f)
+    }
+}
+
+pub struct Value<'a>(pub &'a [u8]);
```

Comment:
`Key` and `Value` can share the same struct and be constructed by different functions.

Reply:
I changed that to one struct `log_wrapper::Value`. Currently they have the same behavior (print one "?"). In the future we could output different format for these two types by adding new fields in `log_wrapper::Value`.